### ISSUE TITLE
Diff independent packages from their own previous tag, not globally most-recent

### ISSUE THREAD

**Issue by Inlustra** [lerna/lerna#1241]
## Expected Behavior

Using independent versioning mode, I'd like Lerna to use the most recent tag associated with that package when using Lerna diff --scope

To elaborate, imagine a simple project structure like the following:

>  PackageA
>  PackageB
>  PackageC

In our CI, we have a pipeline per package, with the intention that breaking 1 of the packages doesn't break the pipeline for the rest of the teams that are working on another package.

One commit could break PackageA and PackageB's pipeline is still able to run.
We do this using the Lerna --scope for each package

## Current Behavior
When Lerna has completed the publish, it would add a git tag such as `PackageA@0.0.1`
However, as we have split each package per pipeline, PackageB will then run but will only check changes since `PackageA@0.0.1` and therefore determine that no changes need to be made

## Possible Solution

We can quite easily fix this by manually add the --since command, using the git tag `PACKAGE_NAME@PACKAGE_VERSION - 1`

### The issue with this: 
A timeline example:
1. PackageA pipeline is triggered and pulls down commit 0
2. PackageA finishes the build and publishes - making this commit number 1
3. PackageB starts up after PackageA and pulls commit 0
  a. The issue at this point is outlined above, but should be fixable using `--since`
4. PackageB finishes the build and publishes.
  a. Git should reject this publish as there have been other commits since this time (I believe)

My Questions:
1. Is there a better way of doing the --since command using lerna?
2. Is there a way to get lerna to pull any changes before publishing (Only to break on merging)

**Comment by evocateur** [comment]
My first reaction to your description is: maybe don't use a monorepo? It sounds terribly convoluted.

In any case, it is fair to request that a scoped diff use independent tags instead of the globally most-recent. Until recently this wasn't possible because lerna used lightweight tags, but it's tractable now.

**Comment by Inlustra** [comment]
Haha, to be honest, these requirements evolved a little later. We started with the mono repo and coupled with gulp it seems to be working well enough

We ended up just custom writing the —since with the package name and current version tag in package.json

**Comment by Inlustra** [comment]
This issue has since been solved.

Taking @evocateur advice and moving this into one pipeline has helped us just fine.
We ended up moving towards a single pipeline for all the libraries, when we do split the teams, we'll end up with 2 pipelines - each team having ownership of their own monorepo of packages. 

> In any case, it is fair to request that a scoped diff use independent tags instead of the globally most-recent. Until recently this wasn't possible because lerna used lightweight tags, but it's tractable now.

This may still be valid but as I seem to be the only one having asked for it, I doubt it's really necessary so I'll close the issue for now 😄
